### DISCUSSION TITLE
How to call Modal.handleUpdate()

### DISCUSSION THREAD

**Discussion by bemyak** [twbs/bootstrap#36580] [Q&A] [answered]
I use Bootstrap `v5.2.0-beta1` and I have a simple modal window:
```html
<div class="modal fade" id="myModal" tabindex="-1" aria-labelledby="myModalLabel" aria-hidden="true">
	<div class="modal-dialog modal-lg modal-dialog-centered">
		<div class="modal-content">
			<div class="modal-body">
                    …
                    </div>
		</div>
	</div>
</div>
```

However, this script doesn't work:
```js
const myModal = document.getElementById('myModal');
myModal.handleUpdate();
```
> `Uncaught TypeError: myModal.handleUpdate is not a function`

I've tried doing this:
```js
const modal = new bootstrap.Modal('#myModal');
modal.handleUpdate();
```
and
```js
const myModal = document.getElementById('myModal');
const modal = new bootstrap.Modal(myModal);
modal.handleUpdate();
```

But it seems to create a new instance of `Modal` that is not linked to the UI element.

What is the proper way of calling the `handleUpdate` function?

**Comment by GeoSot** [comment, marked as answer]
>const myModal = document.getElementById('myModal');
myModal.handleUpdate();

this script rightfully throws an exception as you were trying to call a Modal's method on a HTMLelement

>const modal = new bootstrap.Modal('#myModal');
or   new bootstrap.Modal(myModalElement);

This call should also throw an exception in case the modal is already initialized

 so give a try to 

```js
const input = '#myModal'; //can be HTMLelement or a string selector
const modal = bootstrap.Modal.getOrCreateInstance(input);
modal.handleUpdate();

```

I suppose it will work

**Reply by bemyak** [reply]
This indeed works, thanks!

I was going through `modal.js` code and I haven't found the definition of `getOrCreateInstance` function there :thinking:

**Reply by GeoSot** [reply]
You didn't found it, because it is inherited from its parent class (BaseComponent). ;)

Glad we solve it , so easily
